plugins/emoji/plugin.js

```diff
@@ -0,0 +1,136 @@
+				emoji.getHtmlToInsert = function( item ) {
+					return item.symbol;
+				};
+
+			} );
+
+			editor.on( 'toHtml', function( evt ) {
+				var filter = new CKEDITOR.htmlParser.filter( {
+					text: function( value, element ) {
+						var notConverted = element.getAscendant( getConfigForNotConvertedTags() );
+						if ( notConverted ) {
+							return value;
+						}
+						var hits = value.match( /:[a-zA-Z_-]+?:/g );
+						if ( hits ) {
+							for ( var i = 0; i < hits.length; i++ ) {
+								var item = emoji.find( function( element ) {
+									return element.id === hits[ i ];
+								} );
+								if ( item ) {
+									value = value.replace( item.id, item.symbol );
+								}
+							}
+						}
+						return value;
+					}
+				} );
+				filter.applyTo( evt.data.dataValue );
+			} );
+
+			function getTextTestCallback() {
+				return function( range ) {
+					if ( !range.collapsed || range.startContainer.getAscendant( getConfigForNotConvertedTags() ) ) {
+						return null;
+					}
+					return CKEDITOR.plugins.textMatch.match( range, matchCallback );
+				};
+			}
+
+			function matchCallback( text, offset ) {
+				var left = text.slice( 0, offset ),
+					match = left.match( new RegExp( ':\\S{' + charactersToStart + '}\\S*$' ) );
+
+				if ( !match ) {
+					return null;
+				}
+
+				return { start: match.index, end: offset };
+			}
+
+			function dataCallback( query, range, callback ) {
+				var data = CKEDITOR.tools.array.filter( emoji, function( item ) {
+					return item.id.indexOf( query.slice( 1 ) ) !== -1;
+				} ).sort( function( a, b ) {
+					// Sort at the beginning emoji starts with given query.
+					var emojiName = query.substr( 1 ),
+						isAStartWithEmojiName = a.id.substr( 1, emojiName.length ) === emojiName,
+						isBStartWithEmojiName = b.id.substr( 1, emojiName.length ) === emojiName;
+
+					if ( isAStartWithEmojiName && isBStartWithEmojiName || !isAStartWithEmojiName && !isBStartWithEmojiName ) {
+						return a.id === b.id ? 0 : ( a.id > b.id ? 1 : -1 );
+					} else if ( isAStartWithEmojiName ) {
+						return -1;
+					} else {
+						return 1;
+					}
+				} );
+				callback( data );
+			}
+
+			function getConfigForNotConvertedTags() {
+				var ret = {};
+				for ( var i = 0; i < forbiddenScope.length; i++ ) {
+					ret[ forbiddenScope[ i ] ] = true;
+				}
+				return ret;
+			}
+		}
+	} );
+} )();
+
+
+/**
+ * Array with names of tags where emoji plugin remin inactive. By default emoji are not transformed inside `<pre>` and `<code>` tags
+ *
+ *  ```
+ * 	editor.emoji_preventTransformation = [ 'h1', 'h2' ];
+ * ```
+ * @cfg {Array} [ emoji_preventTransformation = [ 'pre', 'code' ] ]
+ * @member CKEDITOR.config
+ */
+
+/**
+ * Number which defines how many characters is required to start displaying emoji autocomplete.
+ *
+ * ```
+ * 	editor.emoji_numberOfCharactersToStart = 2;
+ * ```
+ * @cfg {Number} [ emoji_numberOfCharactersToStart = 0 ]
```

Comment:
This is a too complex name. Keep it simple. I think it is good idea to reuse the name from mentions `minChars`, so it would be `emoji_minChars`.

It is important to put information whether or not it accounts `:` delimiter.